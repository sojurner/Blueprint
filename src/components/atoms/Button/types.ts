export interface IButtonProps {
  rootProps: {
    styles?: string;
    color?: string;
    disableFocusRipple?: boolean;
    fullWidth?: boolean;
    href?: string;
    size?: 'small' | 'medium' | 'large';
    variant?: 'text' | 'outlined' | 'contained';
  };
}
