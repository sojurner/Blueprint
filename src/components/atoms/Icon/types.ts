export interface ILystIconProps {
  styles?: string;
  iconName: string;
  Icon: Function;
  variant?: IconVariant;
  tooltip: boolean | undefined;
}

export type IconVariant = 'Rounded' | 'Outlined' | 'Sharp' | 'TwoTone' | string;
