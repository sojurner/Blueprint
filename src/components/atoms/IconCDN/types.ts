export interface ILystIconProps {
  styles?: string;
  name: string;
  icon: string;
  variant?: IconVariant;
  tooltip: boolean | undefined;
}

export type IconVariant = 'Rounded' | 'Outlined' | 'Sharp' | 'TwoTone' | string;
