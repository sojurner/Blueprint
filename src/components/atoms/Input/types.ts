export interface ILystInputProps {
  inputProps: IInputProps;
}

interface IInputProps {
  autoComplete?: string;
  autoFocus?: boolean;
  children?: React.ReactNode;
  defaultValue?: unknown;
  disabled?: boolean;
  error?: boolean;
  fullWidth?: boolean;
  helperText?: React.ReactNode;
  id?: string;
  inputRef?: React.Ref<any> | React.RefObject<any>;
  label?: React.ReactNode;
  multiline?: boolean;
  name?: string;
  onChange?: React.ChangeEventHandler<
    HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
  >;
  placeholder?: string;
  required?: boolean;
  rows?: string | number;
  rowsMax?: string | number;
  select?: boolean;
  type?: string;
  value?: unknown;
  styles?: string;
}
