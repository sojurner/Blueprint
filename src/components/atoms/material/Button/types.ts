export interface IButtonProps {
  css: string;
}
