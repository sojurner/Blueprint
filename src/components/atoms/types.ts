export interface IButtonProps {}

export interface IStyleProps {
  className: string;
}

export namespace Hooks {}
