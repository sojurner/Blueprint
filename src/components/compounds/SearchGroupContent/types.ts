import { IFormProps } from '@molecules/Form/types';
import { IListProps } from '@molecules/List/types';

export interface ILystSearchGroupContentProps {
  rootProps: ISearchGroupContentProps;
  listProps: IListProps;
  formProps: IFormProps;
}

export interface ISearchGroupContentProps {
  styles?: string;
}

export interface IAdjustedFormProps extends IFormProps {}

export interface IAdjustedListProps extends IListProps {}
