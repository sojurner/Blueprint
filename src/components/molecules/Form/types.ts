export interface ILystFormProps {
  formProps: {
    inputList: IInputProps[];
    handleChange: Function;
    values: { [key: string]: string };
    styles?: string;
    label?: string;
  };
}

export interface IInputProps {
  inputProps: {
    label: string;
    styles?: string;
    variant: string;
    onChange?: Function;
    name: string;
  };
  iconDetails?:
    | {
        root: 'startAdornment' | 'endAdornment';
        position: string;
        iconName: string;
        variant: string;
      }
    | any;
}
