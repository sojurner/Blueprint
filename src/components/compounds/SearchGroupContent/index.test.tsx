import React from 'react';
import { shallow } from 'enzyme';
import SearchRounded from '@material-ui/icons/SearchRounded';
import SearchOutlined from '@material-ui/icons/SearchOutlined';
import SearchTwoTone from '@material-ui/icons/SearchTwoTone';

import { ILystSearchGroupContentProps } from './types';

import LystSearchGroupContent from '.';

const defaultProps = {
  rootProps: { styles: '' },
  listProps: {
    list: [SearchRounded, SearchOutlined, SearchTwoTone],
    styles: ''
  },
  formProps: {
    inputList: [
      {
        inputProps: {
          label: 'Search Icons',
          name: 'search',
          styles: 'margin: 1rem',
          variant: 'standard'
        },
        iconDetails: {
          root: 'endAdornment',
          position: 'start',
          iconName: 'Search',
          Icon: SearchRounded
        }
      }
    ],
    onChange: () => {},
    label: 'dddd',
    styles: ''
  }
};

const wrap = (props: ILystSearchGroupContentProps) =>
  shallow(<LystSearchGroupContent {...props} />);

describe('LystSearchGroupContent', () => {
  let wrapper = wrap(defaultProps);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a form and List', () => {
    expect(wrapper.children().length).toEqual(2);
  });
});
