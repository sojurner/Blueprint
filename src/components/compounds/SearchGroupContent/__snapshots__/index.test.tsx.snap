// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LystSearchGroupContent should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))>
  <LystForm
    formProps={
      Object {
        "inputList": Array [
          Object {
            "iconDetails": Object {
              "Icon": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "displayName": "SearchRoundedIcon",
                "muiName": "SvgIcon",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "iconName": "Search",
              "position": "start",
              "root": "endAdornment",
            },
            "inputProps": Object {
              "label": "Search Icons",
              "name": "search",
              "styles": "margin: 1rem",
              "variant": "standard",
            },
          },
        ],
        "label": "dddd",
        "onChange": [Function],
        "styles": "",
      }
    }
  />
  <LystList
    listProps={
      Object {
        "list": Array [
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "SearchRoundedIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "SearchOutlinedIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "SearchTwoToneIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
        ],
        "styles": "",
      }
    }
  />
</WithStyles(ForwardRef(Grid))>
`;
